core/src/main/scala/filodb.core/memstore/MemStore.scala

```diff
@@ -139,6 +140,12 @@ trait MemStore extends ChunkSource {
    */
   def indexValues(dataset: DatasetRef, shard: Int, indexName: String, topK: Int = 100): Seq[TermInfo]
 
+  def indexValuesWithFilters(dataset: DatasetRef, shard: Int, filters: Seq[ColumnFilter],
+                          column: String, end: Long, start: Long, limit: Int): List[ZeroCopyUTF8String]
```

Comment:
I believe `column` should be `indexName`,  Also we really need ScalaDoc for this.
Also I think you should just return a `TermInfo`, the API really could almost be collapsed into indexValues (just one API.  Basically, if indexValues had one API and you passed in filters=Nil, then it should be like the first API).

Reply:
@velvia done